### PR TITLE
fshook: Handle short recv or EINTR during recv

### PR DESCRIPTION
**Pull request by sinelaw**

### CODE REVIEW

File: cbits/client.c [hunk 1]
```diff
@@ -105,8 +106,14 @@ static int assert_connection(void)
 bool await_go(void)
 {
     char buf[2];
-    ssize_t rc = recv(assert_connection(), PS(buf), 0);
-    return 2 == rc && !memcmp("GO", PS(buf));
+    uint32_t left = sizeof(buf);
+    while (left > 0) {
+        ssize_t rc = recv(assert_connection(), PS(buf), 0);
```
Comment (da-x) [inline cbits/client.c]: Should we pass `&buf[sizeof(buf) - left], left` here instead of `PS(buf)`?

**Comment by da-x** [comment]
Ah, 'approve' means closing the review. Let's regard it as still open.

File: cbits/client.c (hunk 1 above)
Comment (sinelaw) [inline cbits/client.c]: ugh, yes. I guess what's happening in practice is that it's getting EINTR before the first byte arrives, then it gets both bytes.

**Comment by sinelaw** [comment]
I've push-forced a fixed version, please review

**Pull request merged**